# 00 - Data Review

This notebook validates the processed **Global Superstore** dataset before deeper exploratory analysis.

The goal is to confirm:

- Dataset dimensions.
- Column names.
- Data types.
- Missing values.
- Duplicates.
- Date ranges.
- Main business KPIs.
- Key categorical variables.
- Basic quality checks.

This notebook uses reusable code from the `superstore` package instead of duplicating logic inside the notebook.

In [2]:
#IMPORTS
import pandas as pd
from superstore.data.loader import load_processed_data
from superstore.analytics.kpis import calculate_main_kpis

In [3]:
# Notebook Settings
pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 100)
pd.set_option("display.float_format", "{:,.4f}".format)

In [4]:
# Load processed dataset

df = load_processed_data()

df.head()

,category,city,country,customer_id,customer_name,discount,market,order_date,order_id,order_priority,product_id,product_name,profit,quantity,region,sales,segment,ship_date,ship_mode,shipping_cost,state,sub_category,year,market_group,week_number,order_year,order_month,order_month_name,order_quarter,order_weekday,shipping_days,profit_margin,is_loss,discount_level
0,Office Supplies,Los Angeles,United States,LS-172304,Lycoris Saunders,0.0000,US,2011-01-07,CA-2011-130813,High,OFF-PA-10002005,Xerox 225,9.3312,3,West,19,Consumer,2011-01-09,Second Class,4.3700,California,Paper,2011,North America,2,2011,1,January,1,Friday,2,0.4911,False,no_discount
1,Office Supplies,Los Angeles,United States,MV-174854,Mark Van Huff,0.0000,US,2011-01-21,CA-2011-148614,Medium,OFF-PA-10002893,"Wirebound Service Call Books, 5 1/2"" x 4""",9.2928,2,West,19,Consumer,2011-01-26,Standard Class,0.9400,California,Paper,2011,North America,4,2011,1,January,1,Friday,5,0.4891,False,no_discount
2,Office Supplies,Los Angeles,United States,CS-121304,Chad Sievert,0.0000,US,2011-08-05,CA-2011-118962,Medium,OFF-PA-10000659,"Adams Phone Message Book, Professional, 400 Me...",9.8418,3,West,21,Consumer,2011-08-09,Standard Class,1.8100,California,Paper,2011,North America,32,2011,8,August,3,Friday,4,0.4687,False,no_discount
3,Office Supplies,Los Angeles,United States,CS-121304,Chad Sievert,0.0000,US,2011-08-05,CA-2011-118962,Medium,OFF-PA-10001144,Xerox 1913,53.2608,2,West,111,Consumer,2011-08-09,Standard Class,4.5900,California,Paper,2011,North America,32,2011,8,August,3,Friday,4,0.4798,False,no_discount
4,Office Supplies,Los Angeles,United States,AP-109154,Arthur Prichep,0.0000,US,2011-09-29,CA-2011-146969,High,OFF-PA-10002105,Xerox 223,3.1104,1,West,6,Consumer,2011-10-03,Standard Class,1.3200,California,Paper,2011,North America,40,2011,9,September,3,Thursday,4,0.5184,False,no_discount


In [5]:
# Dataset shape

rows, columns = df.shape

print(f"Rows: {rows:,}")
print(f"Columns: {columns:,}")

Rows: 51,290
Columns: 34


In [6]:
# Columns

df.columns.tolist()

['category',
 'city',
 'country',
 'customer_id',
 'customer_name',
 'discount',
 'market',
 'order_date',
 'order_id',
 'order_priority',
 'product_id',
 'product_name',
 'profit',
 'quantity',
 'region',
 'sales',
 'segment',
 'ship_date',
 'ship_mode',
 'shipping_cost',
 'state',
 'sub_category',
 'year',
 'market_group',
 'week_number',
 'order_year',
 'order_month',
 'order_month_name',
 'order_quarter',
 'order_weekday',
 'shipping_days',
 'profit_margin',
 'is_loss',
 'discount_level']

In [7]:
# Data types
############
# Se busca que las fechas tengan datetime, con eso podemos hacer un analicis de ventas por
# Anio, mes, trimestre, dia de la semana y tiempos de envio
############

df.dtypes

category                       str
city                           str
country                        str
customer_id                    str
customer_name                  str
discount                   float64
market                         str
order_date          datetime64[us]
order_id                       str
order_priority                 str
product_id                     str
product_name                   str
profit                     float64
quantity                     int64
region                         str
sales                        int64
segment                        str
ship_date           datetime64[us]
ship_mode                      str
shipping_cost              float64
state                          str
sub_category                   str
year                         int64
market_group                   str
week_number                  int64
order_year                   int64
order_month                  int64
order_month_name               str
order_quarter       

In [8]:
# Missing values

missing_values = (
    df.isna()
    .sum()
    .sort_values(ascending=False)
)

missing_values[missing_values > 0]

Series([], dtype: int64)

In [9]:
# Missing values summary

total_missing_values = int(df.isna().sum().sum())

print(f"Total missing values: {total_missing_values:,}")

Total missing values: 0


In [10]:
# Duplicate rows

duplicate_rows = int(df.duplicated().sum())
print(f"Duplicated rows: {duplicate_rows:,}")

Duplicated rows: 0


In [11]:
# Date columns validation

df[["order_date", "ship_date"]].dtypes

order_date    datetime64[us]
ship_date     datetime64[us]
dtype: object

In [12]:
# Date range

date_summary = {
    "min_order_date": df["order_date"].min(),
    "max_order_date": df["order_date"].max(),
    "min_ship_date": df["ship_date"].min(),
    "max_ship_date": df["ship_date"].max(),
}

date_summary

{'min_order_date': Timestamp('2011-01-01 00:00:00'),
 'max_order_date': Timestamp('2014-12-31 00:00:00'),
 'min_ship_date': Timestamp('2011-01-03 00:00:00'),
 'max_ship_date': Timestamp('2015-01-07 00:00:00')}

In [13]:
# Check invalid shipping days

invalid_shipping_days = df[df["shipping_days"] < 0]

print(f"Rows with invalid shipping days: {len(invalid_shipping_days)}")

Rows with invalid shipping days: 0


# Interpretacion ejecutiva:

> El negocio vende mucho y si es rentable en general, pero tiene seniales fuertes de riesgos:
> el margen global es de 11.61%, el costo de envio representa 10.70% de las ventas
> y casi el 1 de cada 4 ordenes genera perdida

Eso nos dice que no basta con analizar ventas. Tenemos que analizar

- Rentabilidad
- Descuentos
- Logistica

## KPIs Principales
- total_sales: 12,642,905
- total_profit: 1,467,457.29
- profit_margin: 11.61%
- total_orders: 25,035
- total_customers: 4,873
- average_order_value: 505.01
- average_profit_per_order: 58.62
- average_discount: 14.29%
- total_quantity: 178,312
- total_shipping_cost: 1,352,815.70
- shipping_cost_ratio: 10.70%
- loss_rows: 12,544
- loss_row_rate: 24.46%
- loss_orders: 6,125
- loss_order_rate: 24.46%

In [14]:
# Main KPIs
############
# Primera radiografia del negocio
############
kpis = calculate_main_kpis(df)

kpis

{'total_sales': 12642905.0,
 'total_profit': 1467457.2912800002,
 'profit_margin': 0.11606962887722404,
 'total_orders': 25035,
 'total_customers': 4873,
 'average_order_value': 505.0091871380068,
 'average_profit_per_order': 58.616228930697034,
 'average_discount': 0.14290754533047376,
 'total_quantity': 178312,
 'total_shipping_cost': 1352815.7034,
 'average_shipping_cost': 26.3758179645155,
 'shipping_cost_ratio': 0.10700196698464474,
 'loss_rows': 12544,
 'loss_row_rate': 0.24457009163579646,
 'loss_orders': 6125,
 'loss_order_rate': 0.24465747952865988}

# Senial 1: El negocio es rentable, pero no con margen enorme

Profit margin: 11.61%

Esto significa

> Por cada $100 vendidos, quedan aproximadamente 11.61 de utilidad

No es malo, pero tampoco es un margen alto.

## Hipotesis

Pequenias malas decisiones en:

- Descuentos
- Envios
- Productos

Pueden destruir la utilidad

# Senial 2: El costo logistico es muy relevante

Shipping cost ratio: 10.70%

Esto significa:

> Por cada USD 100 vendidos, aproximadamente USD 10.70 se van en shipping cost.

Comprando con el margen de utilidad de 11.61%, logistica es critica

# Senial 3: Casi 1 de cada 4 ordenes tiene perdida

loss_order_rate: 24.46%

Esto significa:

> Aunque el negocio gana dinero en total, hay muchas ordenes que pierden dinero.

## Hipotesis o pregunta
La siguiente pregunta no es “¿vendemos mucho?”, sino:

- ¿En qué regiones?
- ¿En qué mercados?
- ¿En qué categorías?
- ¿En qué productos?
- ¿Con qué descuentos?
- ¿Con qué métodos de envío?

In [15]:
# Formatted KPIs

formatted_kpis = {
    "Total sales": f"${kpis['total_sales']:,.2f}",
    "Total profit": f"${kpis['total_profit']:,.2f}",
    "Profit margin": f"{kpis['profit_margin']:.2%}",
    "Total orders": f"{kpis['total_orders']:,}",
    "Total customers": f"{kpis['total_customers']:,}",
    "Average order value": f"${kpis['average_order_value']:,.2f}",
    "Average profit per order": f"${kpis['average_profit_per_order']:,.2f}",
    "Average discount": f"{kpis['average_discount']:.2%}",
    "Total quantity": f"{kpis['total_quantity']:,}",
    "Total shipping cost": f"${kpis['total_shipping_cost']:,.2f}",
    "Average shipping cost": f"${kpis['average_shipping_cost']:,.2f}",
    "Shipping cost ratio": f"{kpis['shipping_cost_ratio']:.2%}",
    "Loss rows": f"{kpis['loss_rows']:,}",
    "Loss row rate": f"{kpis['loss_row_rate']:.2%}",
    "Loss orders": f"{kpis['loss_orders']:,}",
    "Loss order rate": f"{kpis['loss_order_rate']:.2%}",
}
formatted_kpis

{'Total sales': '$12,642,905.00',
 'Total profit': '$1,467,457.29',
 'Profit margin': '11.61%',
 'Total orders': '25,035',
 'Total customers': '4,873',
 'Average order value': '$505.01',
 'Average profit per order': '$58.62',
 'Average discount': '14.29%',
 'Total quantity': '178,312',
 'Total shipping cost': '$1,352,815.70',
 'Average shipping cost': '$26.38',
 'Shipping cost ratio': '10.70%',
 'Loss rows': '12,544',
 'Loss row rate': '24.46%',
 'Loss orders': '6,125',
 'Loss order rate': '24.47%'}

In [21]:
# Categorical columns overviews

categorical_columns = [
    "market",
    "market_group",
    "region",
    "country",
    "category",
    "sub_category",
    "segment",
    "ship_mode",
    "order_priority",
    "discount_level",
]

categorical_summary = pd.DataFrame({
    "column": categorical_columns,
    "unique_values": [
        df[column].nunique()
        for column in categorical_columns
    ],
})

categorical_summary

,column,unique_values
0,market,7
1,market_group,6
2,region,13
3,country,147
4,category,3
5,sub_category,17
6,segment,3
7,ship_mode,4
8,order_priority,4
9,discount_level,5


In [22]:
# Unique values for main categorical variables
main_categorical_columns = [
    "market",
    "market_group",
    "category",
    "segment",
    "ship_mode",
    "order_priority",
    "discount_level",
]

for column in main_categorical_columns:
    print(f"\n{column}")
    print(df[column].dropna().unique())


market
<ArrowStringArray>
['US', 'EU', 'LATAM', 'Africa', 'APAC', 'EMEA', 'Canada']
Length: 7, dtype: str

market_group
<ArrowStringArray>
['North America', 'EU', 'LATAM', 'Africa', 'APAC', 'EMEA']
Length: 6, dtype: str

category
<ArrowStringArray>
['Office Supplies', 'Technology', 'Furniture']
Length: 3, dtype: str

segment
<ArrowStringArray>
['Consumer', 'Home Office', 'Corporate']
Length: 3, dtype: str

ship_mode
<ArrowStringArray>
['Second Class', 'Standard Class', 'Same Day', 'First Class']
Length: 4, dtype: str

order_priority
<ArrowStringArray>
['High', 'Medium', 'Critical', 'Low']
Length: 4, dtype: str

discount_level
<ArrowStringArray>
[     'no_discount',  'medium_discount',     'low_discount',
    'high_discount', 'extreme_discount']
Length: 5, dtype: str


In [23]:
# Numeric variables summary

numeric_columns = [
    "sales",
    "profit",
    "quantity",
    "discount",
    "shipping_cost",
    "shipping_days",
    "profit_margin",
]

df[numeric_columns].describe()

,sales,profit,quantity,discount,shipping_cost,shipping_days,profit_margin
count,"51,290.0000","51,290.0000","51,290.0000","51,290.0000","51,290.0000","51,290.0000","51,290.0000"
mean,246.4984,28.6110,3.4765,0.1429,26.3758,3.9694,0.0472
std,487.5672,174.3410,2.2788,0.2123,57.2968,1.7294,0.4668
min,0.0000,"-6,599.9780",1.0000,0.0000,0.0020,0.0000,-4.7278
25%,31.0000,0.0000,2.0000,0.0000,2.6100,3.0000,0.0000
50%,85.0000,9.2400,3.0000,0.0000,7.7900,4.0000,0.1692
75%,251.0000,36.8100,5.0000,0.2000,24.4500,5.0000,0.3331
max,"22,638.0000","8,399.9760",14.0000,0.8500,933.5700,7.0000,0.5828


In [24]:
# Sales and profit sanity check

sanity_check = {
    "sales_less_or_equal_zero": int((df["sales"] <= 0).sum()),
    "profit_negative_rows": int((df["profit"] < 0).sum()),
    "quantity_less_or_equal_zero": int((df["quantity"] <= 0).sum()),
    "discount_less_than_zero": int((df["discount"] < 0).sum()),
    "discount_greater_than_one": int((df["discount"] > 1).sum()),
    "shipping_cost_less_than_zero": int((df["shipping_cost"] < 0).sum()),
}

sanity_check

{'sales_less_or_equal_zero': 1,
 'profit_negative_rows': 12544,
 'quantity_less_or_equal_zero': 0,
 'discount_less_than_zero': 0,
 'discount_greater_than_one': 0,
 'shipping_cost_less_than_zero': 0}

In [25]:
# Dateset period by year

orders_by_year = (
    df.groupby("order_year")
    .agg(
        total_sales=("sales", "sum"),
        total_profit=("profit", "sum"),
        total_orders=("order_id", "nunique"),
    )
    .reset_index()
)

orders_by_year

,order_year,total_sales,total_profit,total_orders
0,2011,2259511,"248,940.8115",4440
1,2012,2677493,"307,415.2791",5343
2,2013,3405860,"406,935.2302",6721
3,2014,4300041,"504,165.9705",8531


In [26]:
# Basic business distribution

business_distribution = {
    "markets": df["market"].nunique(),
    "market_groups": df["market_group"].nunique(),
    "regions": df["region"].nunique(),
    "countries": df["country"].nunique(),
    "categories": df["category"].nunique(),
    "sub_categories": df["sub_category"].nunique(),
    "products": df["product_id"].nunique(),
    "customers": df["customer_id"].nunique(),
    "orders": df["order_id"].nunique(),
}

business_distribution

{'markets': 7,
 'market_groups': 6,
 'regions': 13,
 'countries': 147,
 'categories': 3,
 'sub_categories': 17,
 'products': 10292,
 'customers': 4873,
 'orders': 25035}

# Ventas individuales mas grandes

En ventas grandes no siempre son buenas.

In [16]:
# Top records preview by sales

df.sort_values("sales", ascending=False).head(10)

,category,city,country,customer_id,customer_name,discount,market,order_date,order_id,order_priority,product_id,product_name,profit,quantity,region,sales,segment,ship_date,ship_mode,shipping_cost,state,sub_category,year,market_group,week_number,order_year,order_month,order_month_name,order_quarter,order_weekday,shipping_days,profit_margin,is_loss,discount_level
47257,Technology,Jacksonville,United States,SM-203204,Sean Miller,0.5000,US,2011-03-18,CA-2011-145317,Medium,TEC-MA-10002412,Cisco TelePresence System EX90 Videoconferenci...,"-1,811.0784",6,South,22638,Home Office,2011-03-23,Standard Class,24.2865,Florida,Machines,2011,North America,12,2011,3,March,1,Friday,5,-0.0800,True,high_discount
27727,Technology,Lafayette,United States,TC-209804,Tamara Chand,0.0000,US,2013-10-03,CA-2013-118689,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"8,399.9760",5,Central,17500,Corporate,2013-10-10,Standard Class,349.0700,Indiana,Copiers,2013,North America,40,2013,10,October,4,Thursday,7,0.4800,False,no_discount
28327,Technology,Seattle,United States,RB-193604,Raymond Buch,0.0000,US,2014-03-24,CA-2014-140151,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"6,719.9808",4,West,14000,Consumer,2014-03-26,First Class,20.0005,Washington,Copiers,2014,North America,13,2014,3,March,1,Monday,2,0.4800,False,no_discount
48073,Technology,New York City,United States,TA-213854,Tom Ashbrook,0.2000,US,2014-10-23,CA-2014-127180,High,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"3,919.9888",4,East,11200,Home Office,2014-10-25,First Class,45.9800,New York,Copiers,2014,North America,43,2014,10,October,4,Thursday,2,0.3500,False,medium_discount
27524,Technology,Newark,United States,HL-150404,Hunter Lopez,0.0000,US,2014-11-18,CA-2014-166709,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"5,039.9856",3,East,10500,Consumer,2014-11-23,Standard Class,363.1900,Delaware,Copiers,2014,North America,47,2014,11,November,4,Tuesday,5,0.4800,False,no_discount
3092,Office Supplies,Detroit,United States,AB-101054,Adrian Barton,0.0000,US,2013-12-18,CA-2013-117121,Medium,OFF-BI-10000545,GBC Ibimaster 500 Manual ProClick Binding System,"4,946.3700",13,Central,9893,Consumer,2013-12-22,Standard Class,498.7000,Michigan,Binders,2013,North America,51,2013,12,December,4,Wednesday,4,0.5000,False,no_discount
2722,Office Supplies,Minneapolis,United States,SC-200954,Sanjit Chand,0.0000,US,2011-09-23,CA-2011-116904,Medium,OFF-BI-10001120,Ibico EPK-21 Electric Binding System,"4,630.4755",5,Central,9450,Consumer,2011-09-28,Standard Class,655.6100,Minnesota,Binders,2011,North America,39,2011,9,September,3,Friday,5,0.4900,False,no_discount
27459,Technology,Lakewood,United States,BS-113654,Bill Shonely,0.0000,US,2013-04-17,US-2013-107440,Medium,TEC-MA-10001047,"3D Systems Cube Printer, 2nd Generation, Magenta","2,365.9818",7,East,9100,Corporate,2013-04-21,Standard Class,516.9100,New Jersey,Machines,2013,North America,16,2013,4,April,2,Wednesday,4,0.2600,False,no_discount
28193,Technology,Arlington,United States,SE-201104,Sanjit Engle,0.0000,US,2013-02-02,CA-2013-158841,Critical,TEC-MA-10001127,HP Designjet T520 Inkjet Large Format Printer ...,"2,799.9840",5,South,8750,Consumer,2013-02-04,Second Class,17.8300,Virginia,Machines,2013,North America,5,2013,2,February,1,Saturday,2,0.3200,False,no_discount
46605,Technology,Philadelphia,United States,CC-123704,Christopher Conant,0.4000,US,2013-05-24,CA-2013-143714,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"1,119.9968",4,East,8400,Consumer,2013-05-28,Standard Class,498.0800,Pennsylvania,Copiers,2013,North America,21,2013,5,May,2,Friday,4,0.1333,False,high_discount


# Lineas que mas utilidad generan

In [28]:
# Top records preview by profit

df.sort_values("profit", ascending=False).head(10)

,category,city,country,customer_id,customer_name,discount,market,order_date,order_id,order_priority,product_id,product_name,profit,quantity,region,sales,segment,ship_date,ship_mode,shipping_cost,state,sub_category,year,market_group,week_number,order_year,order_month,order_month_name,order_quarter,order_weekday,shipping_days,profit_margin,is_loss,discount_level
27727,Technology,Lafayette,United States,TC-209804,Tamara Chand,0.0000,US,2013-10-03,CA-2013-118689,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"8,399.9760",5,Central,17500,Corporate,2013-10-10,Standard Class,349.0700,Indiana,Copiers,2013,North America,40,2013,10,October,4,Thursday,7,0.4800,False,no_discount
28327,Technology,Seattle,United States,RB-193604,Raymond Buch,0.0000,US,2014-03-24,CA-2014-140151,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"6,719.9808",4,West,14000,Consumer,2014-03-26,First Class,20.0005,Washington,Copiers,2014,North America,13,2014,3,March,1,Monday,2,0.4800,False,no_discount
27524,Technology,Newark,United States,HL-150404,Hunter Lopez,0.0000,US,2014-11-18,CA-2014-166709,Medium,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"5,039.9856",3,East,10500,Consumer,2014-11-23,Standard Class,363.1900,Delaware,Copiers,2014,North America,47,2014,11,November,4,Tuesday,5,0.4800,False,no_discount
3092,Office Supplies,Detroit,United States,AB-101054,Adrian Barton,0.0000,US,2013-12-18,CA-2013-117121,Medium,OFF-BI-10000545,GBC Ibimaster 500 Manual ProClick Binding System,"4,946.3700",13,Central,9893,Consumer,2013-12-22,Standard Class,498.7000,Michigan,Binders,2013,North America,51,2013,12,December,4,Wednesday,4,0.5000,False,no_discount
2722,Office Supplies,Minneapolis,United States,SC-200954,Sanjit Chand,0.0000,US,2011-09-23,CA-2011-116904,Medium,OFF-BI-10001120,Ibico EPK-21 Electric Binding System,"4,630.4755",5,Central,9450,Consumer,2011-09-28,Standard Class,655.6100,Minnesota,Binders,2011,North America,39,2011,9,September,3,Friday,5,0.4900,False,no_discount
10129,Office Supplies,Prato,Italy,PJ-188352,Patrick Jones,0.0000,EU,2014-09-08,ES-2014-1651774,Low,OFF-AP-10004512,"Hoover Stove, Red","3,979.0800",14,South,7959,Corporate,2014-09-14,Standard Class,778.3200,Tuscany,Appliances,2014,EU,37,2014,9,September,3,Monday,6,0.4999,False,no_discount
48073,Technology,New York City,United States,TA-213854,Tom Ashbrook,0.2000,US,2014-10-23,CA-2014-127180,High,TEC-CO-10004722,Canon imageCLASS 2200 Advanced Copier,"3,919.9888",4,East,11200,Home Office,2014-10-25,First Class,45.9800,New York,Copiers,2014,North America,43,2014,10,October,4,Thursday,2,0.3500,False,medium_discount
2143,Office Supplies,Atlanta,United States,CM-123854,Christopher Martinez,0.0000,US,2012-03-16,CA-2012-145352,Medium,OFF-BI-10003527,Fellowes PB500 Electric Punch Plastic Comb Bin...,"3,177.4750",5,South,6355,Consumer,2012-03-22,Standard Class,581.1200,Georgia,Binders,2012,North America,11,2012,3,March,1,Friday,6,0.5000,False,no_discount
19939,Technology,Fuji,Japan,CA-119651,Carol Adams,0.0000,APAC,2011-02-10,ID-2011-64599,Medium,TEC-PH-10000303,"Samsung Smart Phone, VoIP","2,939.3100",11,North Asia,6999,Corporate,2011-02-15,Standard Class,413.8000,Shizuoka,Phones,2011,APAC,7,2011,2,February,1,Thursday,5,0.4200,False,no_discount
20272,Technology,Kota,India,CA-127751,Cynthia Arntzen,0.0000,APAC,2013-06-12,IN-2013-50809,Medium,TEC-PH-10000169,"Apple Smart Phone, with Caller ID","2,817.9900",9,Central Asia,5752,Consumer,2013-06-17,Standard Class,353.8500,Chhattisgarh,Phones,2013,APAC,24,2013,6,June,2,Wednesday,5,0.4899,False,no_discount


# Las peores perdidas

In [29]:
# Worst records preview by profit

df.sort_values("profit", ascending=True).head(10)

,category,city,country,customer_id,customer_name,discount,market,order_date,order_id,order_priority,product_id,product_name,profit,quantity,region,sales,segment,ship_date,ship_mode,shipping_cost,state,sub_category,year,market_group,week_number,order_year,order_month,order_month_name,order_quarter,order_weekday,shipping_days,profit_margin,is_loss,discount_level
47074,Technology,Lancaster,United States,CS-125054,Cindy Stewart,0.7000,US,2013-11-26,CA-2013-108196,Low,TEC-MA-10000418,Cubify CubeX 3D Printer Double Head Print,"-6,599.9780",5,East,4500,Consumer,2013-12-03,Standard Class,451.6300,Ohio,Machines,2013,North America,48,2013,11,November,4,Tuesday,7,-1.4667,True,extreme_discount
29353,Technology,Bornova,Turkey,DM-33452,Denise Monton,0.6000,EMEA,2013-09-26,TU-2013-9400,Critical,TEC-MOT-10003050,"Motorola Smart Phone, Cordless","-4,088.3760",12,EMEA,3085,Corporate,2013-09-26,Same Day,51.8900,Izmir,Phones,2013,EMEA,39,2013,9,September,3,Thursday,0,-1.3252,True,high_discount
47388,Technology,Burlington,United States,GT-146354,Grant Thornton,0.5000,US,2014-11-05,US-2014-168116,High,TEC-MA-10004125,Cubify CubeX 3D Printer Triple Head Print,"-3,839.9904",4,South,8000,Corporate,2014-11-05,Same Day,674.8200,North Carolina,Machines,2014,North America,45,2014,11,November,4,Wednesday,0,-0.4800,True,high_discount
46422,Office Supplies,San Antonio,United States,LF-171854,Luke Foster,0.8000,US,2011-07-26,CA-2011-169019,Medium,OFF-BI-10004995,GBC DocuBind P400 Electric Binding System,"-3,701.8928",8,Central,2178,Consumer,2011-07-30,Standard Class,172.0700,Texas,Binders,2011,North America,31,2011,7,July,3,Tuesday,4,-1.6997,True,extreme_discount
47414,Technology,Louisville,United States,SR-204254,Sharelle Roach,0.7000,US,2014-04-18,CA-2014-134845,Medium,TEC-MA-10000822,Lexmark MX611dhe Monochrome Laser Printer,"-3,399.9800",5,West,2550,Home Office,2014-04-24,Standard Class,120.3100,Colorado,Machines,2014,North America,16,2014,4,April,2,Friday,6,-1.3333,True,extreme_discount
39499,Office Supplies,Lisbon,Portugal,SS-201402,Saphhira Shifley,0.5000,EU,2013-06-10,IT-2013-3695467,Medium,OFF-AP-10001623,"Hoover Stove, White","-3,059.8200",12,South,3400,Corporate,2013-06-14,Standard Class,286.8700,Lisboa,Appliances,2013,EU,24,2013,6,June,2,Monday,4,-0.8999,True,high_discount
30168,Technology,Lahore,Pakistan,SN-205601,Skye Norling,0.5000,APAC,2013-09-14,ID-2013-12295,Medium,TEC-PH-10002991,"Apple Smart Phone, Full Size","-3,009.4350",11,Central Asia,3499,Home Office,2013-09-14,Same Day,333.1000,Punjab,Phones,2013,APAC,37,2013,9,September,3,Saturday,0,-0.8601,True,high_discount
46876,Office Supplies,Chicago,United States,HG-149654,Henry Goldwyn,0.8000,US,2014-12-08,US-2014-122714,Medium,OFF-BI-10001120,Ibico EPK-21 Electric Binding System,"-2,929.4845",5,Central,1890,Corporate,2014-12-14,Standard Class,129.5100,Illinois,Binders,2014,North America,50,2014,12,December,4,Monday,6,-1.5500,True,extreme_discount
29652,Furniture,Vilnius,Lithuania,JC-61052,Julie Creighton,0.7000,EMEA,2014-01-28,LH-2014-5390,High,FUR-BAR-10003532,"Barricks Conference Table, Rectangular","-2,750.2800",8,EMEA,2171,Corporate,2014-01-30,Second Class,390.0900,Vilnius,Tables,2014,EMEA,5,2014,1,January,1,Tuesday,2,-1.2668,True,extreme_discount
47070,Technology,Newark,United States,NF-183854,Natalie Fritzler,0.7000,US,2012-12-15,CA-2012-147830,High,TEC-MA-10000418,Cubify CubeX 3D Printer Double Head Print,"-2,639.9912",2,East,1800,Consumer,2012-12-18,First Class,219.6200,Ohio,Machines,2012,North America,50,2012,12,December,4,Saturday,3,-1.4667,True,extreme_discount


## Initial validation summary

The processed dataset is ready for exploratory analysis.

Key validation results:

- The dataset contains 51,290 rows and 34 columns.
- There are no duplicated rows.
- There are no missing values in the processed dataset.
- Date columns are correctly loaded as datetime.
- The dataset includes sales, profit, discount, quantity and shipping cost.
- The dataset supports sales, profitability, discount, logistics and customer segment analysis.
- Main KPIs are available through the reusable `calculate_main_kpis` function.

Important initial business signals:

- The business generated more than $12M in sales.
- The global profit margin is around 11.61%.
- Around 24% of rows and orders are loss-making.
- Shipping cost represents around 10.70% of total sales.

This confirms that the dataset is suitable for deeper business analysis.## Initial validation summary

The processed dataset is ready for exploratory analysis.

Key validation results:

- The dataset contains 51,290 rows and 34 columns.
- There are no duplicated rows.
- There are no missing values in the processed dataset.
- Date columns are correctly loaded as datetime.
- The dataset includes sales, profit, discount, quantity and shipping cost.
- The dataset supports sales, profitability, discount, logistics and customer segment analysis.
- Main KPIs are available through the reusable `calculate_main_kpis` function.

Important initial business signals:

- The business generated more than $12M in sales.
- The global profit margin is around 11.61%.
- Around 24% of rows and orders are loss-making.
- Shipping cost represents around 10.70% of total sales.

This confirms that the dataset is suitable for deeper business analysis.